include/charmlite/core/collection_bridge.hpp

```diff
@@ -161,14 +161,15 @@ namespace cmk {
         static entry_id_t receive_status(void)
         {
             using receiver_type = member_fn_t<self_type, data_message<bool>>;
-            return cmk::entry<receiver_type, &self_type::receive_status>();
+            return cmk::entry<static_cast<receiver_type>(
```

Comment:
Is this legal? I thought that casting function pointers was undefined. Why is why is this actually necessary?

Reply:
This cast helps the compiler understand the template types for `self_type::receiver_status`. For cases with overloaded functions, this additional cast helps the compiler decipher the actual function that's required to be called. Consider this basic example for explanation purposes: https://godbolt.org/z/74MzPcnoj

Reply:
To elaborate: The cast here is not changing function pointers. Just helping the compiler understand what the type of the function pointer is in scenarios where the function pointer may be an overload.